Code/GraphMol/ChemReactions/catch_tests.cpp

```diff
@@ -222,4 +222,18 @@ TEST_CASE("negative charge queries. Part of testing changes for github #2604",
     CHECK(nWarnings == 1);
     CHECK(nErrors == 0);
   }
+}
+
+TEST_CASE("GithHub #2954: Reaction Smarts with Dative Bonds not parsed",
+          "[Reaction, Bug]") {
+  
+  SECTION("Rxn Smart Processing with Dative Bond") {
```

Comment:
Please add two additional SECTIONs that show that this also works if the dative bond is in the reactant (reaction SMARTS `[O:1]->[H+]>>[O:1].[H+]`) or in the agents (reaction SMARTS `[O:1][H]>N->[Cu]>[O:1].[H]`)